### PR TITLE
Create ghactions-ci.yml

### PR DESCRIPTION
**Pull request by devenkulkarni**

### CODE REVIEW

**Comment by devenkulkarni** [comment]
@dnoland1 We need to add secrets to Github for env vars:

METAL_AUTH_TOKEN
TF_VAR_project_name

File: .github/workflows/ghactions-ci.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  terraform-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Terraform Format Check
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.5.0
```
Comment (dnoland1) [inline .github/workflows/ghactions-ci.yml:21]: Can we bump this to the latest version of terraform (1.7.5)

File: .github/workflows/ghactions-ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+  pull_request:
+
+jobs:
+  terraform-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (dnoland1) [inline .github/workflows/ghactions-ci.yml:16]: Got a bunch of warnings for this:
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v2. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```
Can we try `actions/checkout@v4`?

File: .github/workflows/ghactions-ci.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      
+      - name: Terraform Format Check
+        uses: hashicorp/setup-terraform@v1
```
Comment (dnoland1) [inline .github/workflows/ghactions-ci.yml:19]: Can we use the latest (v3)? See https://github.com/hashicorp/setup-terraform

**Review by dnoland1 — CHANGES_REQUESTED** [review]
Secrets have been added and a created a dummy commit to re-run the GH Actions. From what I can tell, doesn't look like the terraform commands are running:
image

**Comment by devenkulkarni** [comment]
I have modified the ghactions-ci.yml with couple of changes, hope it works this time.

**Comment by devenkulkarni** [comment]
After optimizing and modifying the ghactions-ci.yml, I see terraform commands running correctly. Please review and let me know if any further changes are required.

Screenshot 2024-04-09 at 4 19 56 PM

**Review by dnoland1 — APPROVED** [review]
Looks good, new version is much better!

Can you create a second PR to remove .drone.yml?

**Pull request merged**